pytorch_lightning/callbacks/progress.py

```diff
@@ -43,8 +42,36 @@ class tqdm(_tqdm):
     Custom tqdm progressbar where we append 0 to floating points/strings to prevent the progress bar from flickering
     """
 
-    @staticmethod
-    def format_num(n) -> str:
+    def __init__(self, *args, **kwargs):
+        self.using_text_io_wrapper = False
+        super().__init__(*args, **kwargs)
+
+        self.using_text_io_wrapper = hasattr(self, "fp") and isinstance(self.fp, io.TextIOWrapper)
+
+    def display(self, msg=None, pos=None):
+        if self.using_text_io_wrapper:
+            if self.n != self.total:
+                msg = " " + self.__repr__()
+                super().display(msg=msg, pos=0)
+            else:
+                super().display()
+                with open(self.fp.name) as fr:
+                    lines = fr.readlines()
+                    lines = [
+                        line
+                        for line in lines
+                        if (not line.startswith(" ") and line != "\n" and "0it " not in line and "-1it " not in line)
+                    ]
+                    self.fp.seek(0)
+                    self.fp.truncate()
+                    for line in lines:
+                        line = line.replace("[A", "")
+                        self.fp.write(line)
+        else:
+            msg = self.__repr__()
+            super().display(msg=msg, pos=0)
+
```

Comment:
is this a fix that could be upstreamed to `tqdm` directly? what guarantees that this override stays in sync with `tqdm`'s API over time?

Reply:
No I don't believe so. 

I would call this more a hack to prevent massive logs while re-directing to a file, but I believe there might smarter way to do this.

Concerning the API, I would say tqdm API is pretty stable right now and this hack is just overriding their display hook.

I am pretty confident they won't be divergence there.